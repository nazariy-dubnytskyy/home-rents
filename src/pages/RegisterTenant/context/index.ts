import * as React from 'react';

type TenantContextType = {
  fullName: string;
  email: string;
  phone: string;
  salaryRange: number;
  completedSteps: number;
  setFullName: (name: string) => void;
  setEmail: (email: string) => void;
  setPhone: (phone: string) => void;
  setSalaryRange: (salaryRange: number) => void;
  setCompletedSteps: (steps: number) => void;
}

const TenantContext = React.createContext<TenantContextType>(undefined);

export default TenantContext;
