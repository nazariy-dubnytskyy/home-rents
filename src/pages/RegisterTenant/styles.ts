import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 40%;
  margin: 0 auto;

  @media(max-width: 1000px) {
    width: 60%;
  }

  @media(max-width: 768px) {
    width: 90%;
    padding: 0 ${props => props.theme.spaceUnit.two};
  }
`;

export const ProgressBarWrapper = styled.div`
  padding-bottom: ${props => props.theme.spaceUnit.three};
`;

export const Title = styled.div`
  font-size: ${props => props.theme.fontSizes.xlarge};
  padding-top: ${props => props.theme.spaceUnit.three};
  padding-bottom: ${props => props.theme.spaceUnit.four};
`;

export const ButtonsWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  padding-top: ${props => props.theme.spaceUnit.three};
`;