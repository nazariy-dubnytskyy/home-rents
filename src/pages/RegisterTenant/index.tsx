import * as React from 'react';
import { Switch, Route } from 'react-router-dom';

import UserFullName from './components/UserFullName';
import Email from './components/Email';
import PhoneNumber from './components/PhoneNumber';
import Salary from './components/Salary';
import TenantProvider from './context/TenantProvider';
import RegisterTenantSummary from './components/RegisterTenantSummary';
import { Wrapper } from './styles';
import Progress from './components/Progress';

function RegisterTenant() {
  return (
    <TenantProvider>
      <Wrapper>
        <Progress />
        <Switch>
            <Route path="/register-tenant/name" component={UserFullName} />
            <Route path="/register-tenant/email" component={Email} />
            <Route path="/register-tenant/phone" component={PhoneNumber} />
            <Route path="/register-tenant/salary" component={Salary} />
            <Route path="/register-tenant/summary" component={RegisterTenantSummary} />
        </Switch>
      </Wrapper>
    </TenantProvider>
  )
}

export default RegisterTenant;