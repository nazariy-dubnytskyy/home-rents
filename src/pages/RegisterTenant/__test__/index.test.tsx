import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { render } from '../../../utils/test-utils';
import RegisterTenant from '..';

describe('Home page', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <RegisterTenant />
      </BrowserRouter>
    );


    expect(container.firstChild).toMatchSnapshot();
  });
});