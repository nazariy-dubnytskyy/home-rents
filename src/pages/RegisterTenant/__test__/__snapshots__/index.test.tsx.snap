// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div
  class="sc-AxirZ ewCydr"
>
  <div
    class="sc-AxiKw jhjbZV"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="sc-fzpans eQbzcd"
      role="progressbar"
      value="0"
    />
  </div>
</div>
`;
